src/Microsoft.SqlTools.ServiceLayer/Scripting/ScriptingScriptOperation.cs

```diff
@@ -218,16 +218,46 @@ private SqlScriptPublishModel BuildPublishModel()
                     "Scripting object count {0}, objects: {1}",
                     selectedObjects.Count(),
                     string.Join(", ", selectedObjects)));
-
+            
+            string server = GetServerNameFromLiveInstance(this.Parameters.ConnectionString);
             string database = new SqlConnectionStringBuilder(this.Parameters.ConnectionString).InitialCatalog;
+
+
             foreach (ScriptingObject scriptingObject in selectedObjects)
             {
-                publishModel.SelectedObjects.Add(scriptingObject.ToUrn(database));
+                publishModel.SelectedObjects.Add(scriptingObject.ToUrn(server, database));
             }
 
             return publishModel;
         }
 
+        private string GetServerNameFromLiveInstance(string connectionString)
+        {
+            string serverName = string.Empty;
+            try
+            {
+                using(SqlConnection connection = new SqlConnection(connectionString))
+                {
+                    connection.Open();
+                    
+                    using(SqlCommand cmd = connection.CreateCommand())
+                    {
+                        cmd.CommandText = "SELECT @@servername";
+
+                        object retValue = cmd.ExecuteScalar();  
+                        serverName = retValue as string;
+                    }
+                }
+            }
+            catch (Exception e)
+            {
+                Logger.Write(LogLevel.Error, string.Format("Error during SQL query for server name. Message: {0}", e.Message));
+                throw e;
```

Comment:
>e [](start = 22, length = 1)

The 'throw e' statement will reset the callstack. I think you just want 'throw'. However, I think we should remove the try catch - the call stack will be pretty clear in the log so I don't think we need to log a message.